src/sha2.rs

```diff
@@ -43,13 +14,13 @@ pub struct Sha256 {
 impl Sha256 {
```

Comment:
shouldn't you specify the trait you're implementing here?

Reply:
I thought `Sha256` was going to implement some simple trait, but it turns out to be a type of [`CoreWrapper`](https://docs.rs/digest/0.10.3/digest/core_api/struct.CoreWrapper.html), which implements a ton of things. So I'm just going to make the two options duck-type compatible by implement the three functions we need in `more_ops.rs`.